forge/lib/tiler.py

```diff
@@ -80,11 +79,43 @@ def worker(job):
         db = DB('database.cfg')
         session = sessionmaker()(bind=db.userEngine)
 
+        # Get the model according to the zoom level
         model = loc_models[str(tileXYZ[2])]
+
+        # Get the interpolated point at the 4 corners
+        # 0: (minX, minY), 1: (minX, maxY), 2: (maxX, maxY), 3: (maxX, minY)
+        p_0 = (bounds[0], bounds[1], 0)
+        p_1 = (bounds[0], bounds[3], 0)
+        p_2 = (bounds[2], bounds[3], 0)
+        p_3 = (bounds[2], bounds[1], 0)
+        pts = [p_0, p_1, p_2, p_3]
+        cornerPts = {}
+        # Get the height of the corner points as postgis cannot properly clip
+        # a polygon
+        for pt in pts:
+            query = session.query(model.id, model.interpolateHeightOnPlane(pt, model.the_geom).label('p')).filter(
+                and_(model.bboxIntersects(bounds), model.pointIntersects(pt))
+            )
+            for q in query:
+                cornerPts[q.id] = list(to_shape(WKBElement(q.p)).coords)
```

Comment:
Might be worth a try (though I wouldn't know directly how in polar coordinates...), as it would scale probably better if we use a central db instance in the future.